parcel.go

```diff
@@ -14,47 +16,108 @@ func NewParcelStore(db *sql.DB) ParcelStore {
 
 func (s ParcelStore) Add(p Parcel) (int, error) {
 	// реализуйте добавление строки в таблицу parcel, используйте данные из переменной p
-
-	// верните идентификатор последней добавленной записи
-	return 0, nil
+	res, err := s.db.Exec("INSERT INTO parcel (client, status, address, created_at) VALUES (:client, :status, :address, :created_at)",
+		sql.Named("client", p.Client),
+		sql.Named("status", p.Status),
+		sql.Named("address", p.Address),
+		sql.Named("created_at", p.CreatedAt),
+	)
+	if err != nil {
+		return 0, err
+	}
+	number, err := res.LastInsertId()
+	if err != nil {
+		return 0, err
+	}
+	return int(number), nil
 }
 
 func (s ParcelStore) Get(number int) (Parcel, error) {
 	// реализуйте чтение строки по заданному number
 	// здесь из таблицы должна вернуться только одна строка
-
-	// заполните объект Parcel данными из таблицы
+	row := s.db.QueryRow("SELECT number, client, status, address, created_at FROM parcel WHERE number = ?", number)
 	p := Parcel{}
-
+	err := row.Scan(&p.Number, &p.Client, &p.Status, &p.Address, &p.CreatedAt)
+	if err != nil {
+		return Parcel{}, err
+	}
 	return p, nil
 }
 
 func (s ParcelStore) GetByClient(client int) ([]Parcel, error) {
 	// реализуйте чтение строк из таблицы parcel по заданному client
 	// здесь из таблицы может вернуться несколько строк
+	rows, err := s.db.Query("SELECT number, client, status, address, created_at FROM parcel WHERE client = ?", client)
+	if err != nil {
+		fmt.Println(err)
+		return []Parcel{}, err
+	}
+	defer rows.Close()
 
-	// заполните срез Parcel данными из таблицы
 	var res []Parcel
-
+	for rows.Next() {
+		p := Parcel{}
+		err := rows.Scan(&p.Number, &p.Client, &p.Status, &p.Address, &p.CreatedAt)
+		if err != nil {
+			fmt.Println(err)
+			return []Parcel{}, err
+		}
+		res = append(res, p)
+	}
+	err = rows.Err()
+	if err != nil {
+		return []Parcel{}, err
+	}
 	return res, nil
 }
 
 func (s ParcelStore) SetStatus(number int, status string) error {
 	// реализуйте обновление статуса в таблице parcel
-
+	_, err := s.db.Exec("UPDATE parcel SET status = :status WHERE number = :number",
+		sql.Named("status", status),
+		sql.Named("number", number),
+	)
+	if err != nil {
+		return err
+	}
 	return nil
 }
 
 func (s ParcelStore) SetAddress(number int, address string) error {
 	// реализуйте обновление адреса в таблице parcel
 	// менять адрес можно только если значение статуса registered
-
-	return nil
+	parcel, err := s.Get(number)
+	if err != nil {
+		return err
+	}
+	if parcel.Status == ParcelStatusRegistered {
+		_, err := s.db.Exec("UPDATE parcel SET address = :address WHERE number = :number",
+			sql.Named("address", address),
+			sql.Named("number", number),
+		)
+		if err != nil {
+			return err
+		}
+		return nil
+	} else {
```

Comment:
Так как в if есть возрат, то else не нужен
```suggestion
	}
```